libraries/provider_slave_grants.rb

```diff
@@ -0,0 +1,45 @@
+class Chef
+  class Provider
+    # Used to setup permissions for MySQL replication user on master server
+    class MysqlmSlaveGrants < Chef::Provider::LWRPBase
+      use_inline_resources if defined?(use_inline_resources)
+
+      def whyrun_supported?
+        true
+      end
+
+      def action_create
+        grant_slave_privs
```

Comment:
It would be interesting to see what that looks like. It feels like the 'right way' to do that would be a ruby mysql gem call to examine grants (as opposed to shelling out). Should we be using a more ruby-friendly / standard way of accessing mysql than shelling out? I assume the upstream database and mysql cookbooks include the mysql gem installation for the same purpose.